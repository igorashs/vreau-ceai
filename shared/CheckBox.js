import styled from 'styled-components';
import { Label } from './Label';

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  gap: 7px;
`;

const StyledInput = styled.input`
  margin: 0;
  width: 16px;
  height: 16px;
  order: -1;

  cursor: pointer;
`;

export const CheckBox = ({ name, label, error, passRef, id = name }) => {
  return (
    <Wrapper>
      <Label htmlFor={id} error={error} success>
        {error || label}
      </Label>

      <StyledInput id={id} ref={passRef} name={name} type="checkbox" />
    </Wrapper>
  );
};
