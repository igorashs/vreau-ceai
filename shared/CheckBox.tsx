import styled from 'styled-components';
import { Label } from './Label';

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  gap: 7px;
`;

const StyledInput = styled.input`
  margin: 0;
  width: 16px;
  height: 16px;
  order: -1;

  cursor: pointer;
`;

interface CheckBoxProps {
  name: string;
  label: string;
  error?: boolean;
  passRef?:
    | ((instance: HTMLInputElement) => void)
    | React.RefObject<HTMLInputElement>;
  id?: string;
}

const CheckBox = ({
  name,
  label,
  error = false,
  passRef = null,
  id = name,
}: CheckBoxProps) => {
  return (
    <Wrapper>
      <Label htmlFor={id} error={error} success>
        {error || label}
      </Label>

      <StyledInput id={id} ref={passRef} name={name} type="checkbox" />
    </Wrapper>
  );
};

export default CheckBox;
