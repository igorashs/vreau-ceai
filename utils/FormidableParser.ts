import * as validator from '@/utils/validator';
import FormidableNameSpace, { Files, File, IncomingForm } from 'formidable';
import { NextApiRequest } from 'next';
import { Formidable, ProductFields } from 'types';
import { productMessages } from './validator/schemas/product';
import { NextApiSessionRequest } from './withSession';

type ParsedData = {
  fields: Partial<ProductFields>;
  files: Files & { src?: File };
};

type HandleProductForm = (
  req: NextApiRequest | NextApiSessionRequest,
  options: Partial<FormidableNameSpace.Options>,
  allowedFileTypes: string[],
) => Promise<ParsedData>;

const handleProductForm: HandleProductForm = async (
  req,
  options,
  allowedFileTypes,
) => {
  const data: ParsedData = await new Promise((resolve, reject) => {
    const form: Formidable = new IncomingForm(options);

    form.onPart = (part) => {
      if (part.filename) {
        // get file extension
        const fileType = part.mime?.split('/').pop() ?? '';

        // validate allowed file types
        if (!allowedFileTypes.includes(fileType)) {
          form.emit(
            'error',
            validator.createValidationError({
              message: productMessages.src.invalid,
              key: 'src',
            }),
          );
        }
      }

      form.handlePart(part);
    };

    form.parse(req, (err, fields, files) => {
      if (err) {
        // custom error on maxFileSize
        if (err.message.includes('maxFileSize')) {
          reject(
            validator.createValidationError({
              message: productMessages.src.max,
              key: 'src',
            }),
          );
        } else {
          reject(err);
        }
      }

      resolve({ fields, files });
    });
  });

  return data;
};

export default {
  handleProductForm,
};
