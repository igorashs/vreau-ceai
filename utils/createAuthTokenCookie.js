import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createAuthTokenCookie = (payload) => {
  const authToken = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  return cookie.serialize('auth-token', authToken, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });
};
