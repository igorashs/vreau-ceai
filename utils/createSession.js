import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createSession = (payload) => {
  const token = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  const [header, jwtPayload, signature] = token.split('.');
  const session = header + '.' + jwtPayload;

  const sessionCookie = cookie.serialize('session', session, {
    secure: process.env.NODE_ENV !== 'development',
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  const authCookie = cookie.serialize('auth', signature, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  return [sessionCookie, authCookie];
};
