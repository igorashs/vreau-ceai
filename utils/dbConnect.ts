import mongoose from 'mongoose';

const dbConnect = async () => {
  if (mongoose.connection.readyState >= 1) {
    return;
  }

  await mongoose.connect(process.env.MONGODB_URI as string, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useCreateIndex: true,
    useFindAndModify: false,
  });
};

export default dbConnect;
