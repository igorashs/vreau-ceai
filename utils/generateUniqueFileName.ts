import { nanoid } from 'nanoid';

const generateUniqueFileName = (fileName: string) => {
  // grab file extension
  const matches = fileName?.match(/\..+$/);

  // create new file name
  const uniqueFileName = nanoid() + (matches && matches[0]);

  return uniqueFileName;
};

export default generateUniqueFileName;
