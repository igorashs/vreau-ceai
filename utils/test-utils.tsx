import { render, RenderOptions } from '@testing-library/react';
import GlobalStyle from 'GlobalStyle';

const Wrapper: React.FC = ({ children }) => (
  <>
    <GlobalStyle />
    {children}
  </>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
