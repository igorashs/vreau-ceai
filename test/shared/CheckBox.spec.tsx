import { render } from '@/utils/test-utils';
import CheckBox from '@/shared/CheckBox';

describe('CheckBox', () => {
  it('renders correctly', () => {
    const name = 'checkbox';
    const label = 'checkbox label';

    const { container, getByText, getByRole } = render(
      <CheckBox name={name} label={label} />,
    );

    const checkbox = getByRole('checkbox');

    expect(checkbox).toHaveAttribute('name', name);
    expect(checkbox).toHaveAttribute('id', name);
    expect(getByText(label)).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('error prop', () => {
    it('renders with error message', () => {
      const name = 'checkbox';
      const label = 'checkbox label';
      const errorMsg = 'error';

      const { getByText, queryByText } = render(
        <CheckBox name={name} label={label} error={errorMsg} />,
      );

      expect(queryByText(label)).not.toBeInTheDocument();
      expect(getByText(errorMsg)).toBeVisible();
    });
  });
});
