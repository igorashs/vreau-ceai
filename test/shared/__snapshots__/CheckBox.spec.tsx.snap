// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders correctly 1`] = `
.c1 {
  font-weight: 500;
  color: var(--accent-text-dark);
  font-weight: 400;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 7px;
}

.c2 {
  margin: 0;
  width: 16px;
  height: 16px;
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  cursor: pointer;
}

<div
  class="c0"
>
  <label
    class="c1"
    for="checkbox"
  >
    checkbox label
  </label>
  <input
    class="c2"
    id="checkbox"
    name="checkbox"
    type="checkbox"
  />
</div>
`;
